### ISSUE TITLE
Compilation of the library under Anaconda (CI)

### ISSUE THREAD

**Issue by Chris00** [Chris00/rust-matplotlib#4]
The continuous integration test for Anconda [is failing](https://github.com/Chris00/rust-matplotlib/actions/runs/8606410930/job/23929720746#step:14:1).  @LCrossman If you have time (and since you use Anaconda — I don't), 
- is it the same on your machine?
- any idea of the cause of the problem (and possibly a fix)?

**Comment by LCrossman** [comment]
I *think* it's a mac osx arm64 error call because in Test (macos) step, 5th step, the call is to run rustup target add x86_64-apple-darwin 
Then in failing step, the call is not specified arm64 or x86_64 through rosetta, could it be calling arm64 Mac version but with x86_64 expected?

**Comment by Chris00** [comment]
When installing Anaconda, the reported platform is [`oxs-64`](https://github.com/Chris00/rust-matplotlib/actions/runs/8739686965/job/23981557259#step:7:8) and the Rust toolchain is [`stable-x86_64-apple-darwin unchanged`](https://github.com/Chris00/rust-matplotlib/actions/runs/8739686965/job/23981557259#step:8:16).  If there is a mismatch, it's origin is not clear to me.  Also, so bad the missing symbol is not printed out...

**Comment by LCrossman** [comment]
I have it as working on arm64 rustc 1.76 as long as the 
```export DYLD_LIBRARY_PATH=$CONDA_PREFIX/lib```
is called.  
Under x86_64 I have rustc 1.71 it works with the simple example but I am getting this polars arrow error with the flower example:
[Polars arrow error](https://stackoverflow.com/questions/77894678/if-and-else-have-incompatible-types-in-polars-arrow-0-37-0)

**Comment by Chris00** [comment]
I added
```
        export DYLD_PRINT_LIBRARIES=1
        export DYLD_PRINT_APIS=1
        export DYLD_PRINT_WARNINGS=1
```
to [print additional information](https://github.com/Chris00/rust-matplotlib/actions/runs/8744585633/job/23997802564#step:13:614) but I still don't understand the cause of the problem.

Glad that it works for you.

For the Polars problem, [you may want to add your voice](https://github.com/pola-rs/polars/issues/14134).

**Comment by Chris00** [comment]
If I specify arm-86, I get the error [error: command failed: 'cargo': Bad CPU type in executable (os error 86)](https://github.com/Chris00/rust-matplotlib/actions/runs/8744946074/job/23998929849#step:13:9).

**Comment by Chris00** [comment]
This type of error is [referred to in the Pyo3 doc](https://pyo3.rs/v0.21.0/faq#i-cant-run-cargo-test-or-i-cant-build-in-a-cargo-workspace-im-having-linker-issues-like-symbol-not-found-or-undefined-reference-to-_pyexc_systemerror)!  But it does not seem to be related to our problem...

**Comment by Chris00** [comment]
I've added a [MacOS aarch64 worker](https://github.com/Chris00/rust-matplotlib/actions/runs/8756202997/job/24032105147) to the CI and it works fine.

**Comment by Chris00** [comment]
I also added anaconda on arm64... and it [fails the same way as for amd64](https://github.com/Chris00/rust-matplotlib/actions/runs/8756402848/job/24032762744#step:14:13).  For now, I allowed it to fail. 😢

**Comment by LCrossman** [comment]
Not sure of the issue...could try miniconda instead which is lighter and I am using.
[Miniconda](https://docs.anaconda.com/free/miniconda/)

**Comment by Chris00** [comment]
Thanks for your reply.  That's [what I am using for MacOS arm64](https://github.com/Chris00/rust-matplotlib/blob/dev/.github/workflows/rust.yml#L63) because the worker (provided by [FlyCI](https://github.com/apps/flyci-prod)) does not come with Anaconda installed.  Unfortunately, [the end result is the same](https://github.com/Chris00/rust-matplotlib/actions/runs/8756402848/job/24032762744#step:14:13).  It's a really frustrating error as it [used to work](https://github.com/Chris00/rust-matplotlib/actions/runs/8265638331/job/22611745824#step:14:152) and it is not clear what changed...

**Comment by LCrossman** [comment]
Yes I am not clear on the issue, I think we will need to post on conda website, because it works locally.
Or use brew instead?
One other thing which I guess is probably unrelated is that it has worked locally and save the file but it does also open and close a plotting window too quickly to look at the plot.

**Comment by Chris00** [comment]
Now I have [an error](https://github.com/Chris00/rust-matplotlib/actions/runs/8930665951/job/24531219398#step:12:1) that resembles to [one of `curl-sys`](https://github.com/alexcrichton/curl-rust/issues/524) but I can't find its cause... :(

**Comment by Chris00** [comment]
See also https://github.com/rust-lang/cargo/issues/12670

**Comment by Chris00** [comment]
The problem was the `DYLD_LIBRARY_PATH` itself.  Indeed this directory contains many other libraries which where chosen instead of the default ones. Some of these where not linked with MacOSX system libraries whence the exception.

**Comment by LCrossman** [comment]
DYLD_LIBRARY_PATH specification was indeed all that was needed to fix locally

**Comment by Chris00** [comment]
Unfortunately, that is also was was causing the problem for the CI.  Let's hope it is fixed [at the PyO3 level](https://github.com/PyO3/pyo3/issues/4155) — or at the [anaconda level](https://github.com/ContinuumIO/anaconda-issues/issues/11439).  This issue is already 4.5 years old...

**Comment by Chris00** [comment]
Well, all tests (including using anaconda) [now pass](https://github.com/Chris00/rust-matplotlib/actions/runs/12482380124) but they depend on a "link trick":
```
mkdir -p target/debug/deps
ln -s $CONDA_PREFIX/lib/libpython* target/debug/deps
```
